utilities/update_version.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+# This script updates the READMEs with the latest non-SNAPSHOT version number and increments the SNAPSHOT 
+# version number in the pom.xml files.
+
+# Precondition: the pom.xml is at a "-SNAPSHOT" version before running this script.  
+# Argument (optional):
+#   $1: new version number for pom.xml files (do not include -SNAPSHOT, that is done automatically).
+#   Providing no argument defaults to incrementing revision number from x.y.z-SNAPSHOT to x.y.z+1-SNAPSHOT
+
+# Example: Suppose that before running this script, the pom reads 7.8.9-SNAPSHOT.  This script will replace 
+# all occurrences of #.#.# with 7.8.9 in the README files and the first occurrence of 7.8.9-SNAPSHOT with 
+# 7.8.10-SNAPSHOT in the pom.xml files.
+
+# Get the previous maven project version.
+CURRENT_SNAPSHOT=$(mvn org.apache.maven.plugins:maven-help-plugin:2.1.1:evaluate -Dexpression=project.version | grep -Ev '(^\[|\w+:)')
+RELEASED_VERSION=${CURRENT_SNAPSHOT%%"-"*}
+DEFAULT_UPDATE="${RELEASED_VERSION%.*}.$((${RELEASED_VERSION##*.}+1))"
+NEW_SNAPSHOT_VERSION=${1:-$DEFAULT_UPDATE}-SNAPSHOT
+
+echo "Changing version to $RELEASED_VERSION in README files"
+echo "Changing version to $NEW_SNAPSHOT_VERSION in pom.xml files"
+
+sed -ri "s/[0-9][0-9]*.[0-9][0-9]*.[0-9][0-9]*/$RELEASED_VERSION/g" README.md
```

Comment:
Sounds good for the url-redirect.
Even <version>x.y.z</version> is tricky as a version could appear in other places in the pom (such as plugin version, see our pom for example). However, I am less worried about the poms as for now we are still going to manually update them but we will need something better when we let Travis do that.